runtime/parachains/src/configuration/migration.rs

```diff
@@ -16,10 +16,319 @@
 
 //! A module that is responsible for migration of storage.
 
-use frame_support::traits::StorageVersion;
+use crate::configuration::{self, Config, Pallet, Store, MAX_POV_SIZE};
+use frame_support::{
+	pallet_prelude::*,
+	traits::StorageVersion,
+	weights::{OldWeight, Weight},
+};
+use frame_system::pallet_prelude::BlockNumberFor;
 
 /// The current storage version.
 ///
 /// v0-v1: <https://github.com/paritytech/polkadot/pull/3575>
 /// v1-v2: <https://github.com/paritytech/polkadot/pull/4420>
-pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
+/// v2-v3: <https://github.com/paritytech/polkadot/pull/6091>
+pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(3);
+
+/// Migrates the pallet storage to the most recent version, checking and setting the `StorageVersion`.
+pub fn migrate_to_latest<T: Config>() -> Weight {
+	let mut weight = Weight::zero();
+	if StorageVersion::get::<Pallet<T>>() == 2 {
+		weight += migrate_to_v3::<T>();
+		StorageVersion::new(3).put::<Pallet<T>>();
+	}
+	weight
+}
+
+mod v2 {
+	use super::*;
+	use primitives::v2::{Balance, SessionIndex};
+
+	// Copied over from configuration.rs @ de9e147695b9f1be8bd44e07861a31e483c8343a and removed
+	// all the comments, and changed the Weight struct to OldWeight
+	#[derive(parity_scale_codec::Encode, parity_scale_codec::Decode, Debug)]
+	pub struct HostConfiguration<BlockNumber> {
```

Comment:
How about making the struct generic and doing something like `HostConfig<BlockNumber, WeightV2>`?  
The code is difficult to review like this.

Reply:
I'm not sure what you're suggesting here? How would making the `Weight` generic help?

Reply:
So that you dont have to copy-paste the type definition from `configurations.rs`.